File: backend-java/src/main/java/com/empire/employeefinder/service/JobService.java
package com.empire.employeefinder.service;

import com.empire.employeefinder.dto.response.JobPositionResponseDto;
import com.empire.employeefinder.dto.response.JobTypeWithPositionsResponseDto;
import com.empire.employeefinder.model.JobType;
import com.empire.employeefinder.model.enums.Gender;
import com.empire.employeefinder.model.enums.Location;

import java.util.List;

public interface JobService {

    List<JobPositionResponseDto> getAllJobPositionsFromNewAndSelectedEmployees();

    List<JobPositionResponseDto> getJobPositionsByJobType(Long jobTypeId);

    List<JobPositionResponseDto> getJobPositionsByJobTypeFromNewAndSelectedEmployees(Long jobTypeId);

    List<JobTypeWithPositionsResponseDto> getAllJobTypesWithPositions();

    List<JobTypeWithPositionsResponseDto> getJobTypesWithPositionsByGender(Gender gender);

    List<JobTypeWithPositionsResponseDto> getJobTypesWithPositionsByLocation(Location location);

    List<JobTypeWithPositionsResponseDto> getJobTypesWithPositionsByGenderAndLocation(Gender gender, Location location);

    List<JobType> getAllJobTypes();

    List<JobType> getJobTypesFromNewAndSelectedEmployees();

    List<JobPositionResponseDto> getAllJobPositions();
}

File: frontend-react/src/services/JobService.js
import axios from 'axios';

const JOBS_URL = "http://localhost:8081/api/v1/jobs"

class JobService {

    getAllJobTypes = () => {
        return axios.get(`${JOBS_URL}/job-types`);
    }

    getJobTypesFromNewAndSelectedEmployees = () => {
        return axios.get(`${JOBS_URL}/job-types/from-employees`);
    }

    getAllJobPositions = () => {
        return axios.get(`${JOBS_URL}/job-positions`);
    }

    getAllJobPositionsFromNewAndSelectedEmployees = () => {
        return axios.get(`${JOBS_URL}/job-positions/from-employees`);
    }

    getJobPositionsByJobTypeId = (jobTypeId) => {
        return axios.get(`${JOBS_URL}/job-types-positions?jobTypeId=${jobTypeId}`);
    }

    getJobPositionsByJobTypeIdFromNewAndSelectedEmployees = (jobTypeId) => {
        return axios.get(`${JOBS_URL}/job-types-positions/from-employees?jobTypeId=${jobTypeId}`);
    }

    getAllJobTypesWithPositions = () => {
        return axios.get(`${JOBS_URL}/job-types-positions/all`);
    }

    getJobTypesWithPositionsByGender = (gender) => {
        return axios.get(`${JOBS_URL}/job-types-positions/gender?gender=${gender}`);
    }

    getJobTypesWithPositionsByLocation = (location) => {
        return axios.get(`${JOBS_URL}/job-types-positions/location?location=${location}`);
    }

    getJobTypesWithPositionsByGenderAndLocation = (gender, location) => {
        return axios.get(`${JOBS_URL}/job-types-positions/gender-location?location=${location}&gender=${gender}`);
    }
}

export default new JobService();